agent/stats/engine.go

```diff
@@ -395,6 +418,23 @@ func (engine *DockerStatsEngine) getContainerMetricsForTask(taskArn string) ([]*
 	return containerMetrics, nil
 }
 
+func (engine *DockerStatsEngine) doRemoveContainer(container *StatsContainer, taskArn string) {
+	container.StopStatsCollection()
+	dockerID := container.containerMetadata.DockerID
+	delete(engine.tasksToContainers[taskArn], dockerID)
+	seelog.Debugf("Deleted container from tasks, id: %s", dockerID)
+
+	if len(engine.tasksToContainers[taskArn]) == 0 {
```

Comment:
Looks like you're only handling the case where there are no containers in the task.  Do you need to handle the case where there are containers?